test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java

```diff
@@ -286,13 +286,12 @@ public static InstanceConfig generate(int nodeNum,
 
     public InstanceConfig forVersion(Versions.Major major)
     {
-        switch (major)
-        {
-            case v4: return this;
-            default: return new InstanceConfig(this)
+        if (Versions.Major.v40.compareTo(major) <= 0)
+            return this;
+        else
+            return new InstanceConfig(this)
```

Comment:
We could actually add a small comment on compatibility of `seeds` between the versions here. Definitely not a problem with a patch itself, just can use it as an opportunity to improve, while we have spotted it.